### PR TITLE
Update terminology from credentialfree to passwordless.

### PR DESCRIPTION
**Pull request by backwind1233**
This PR focus on updating terminology from credentialfree to passwordless.
Use this [PR](https://github.com/Azure/azure-sdk-for-java/pull/30853) to improve the readme.

### CODE REVIEW

**Comment by backwind1233** [comment]
https://github.com/Azure/azure-sdk-for-java/issues/30787

File: sdk/boms/spring-cloud-azure-dependencies/pom.xml [hunk 1]
```diff
@@ -55,6 +55,11 @@
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <dependency>
```
Comment (saragluna) [inline sdk/boms/spring-cloud-azure-dependencies/pom.xml:58]: Please add some description here to explain why we are adding this dependency.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jdbc/JdbcPropertiesBeanPostProcessor.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import static com.azure.spring.cloud.service.implementation.identity.credential.provider.SpringTokenCredentialProvider.CREDENTIAL_FREE_TOKEN_BEAN_NAME;
+import static com.azure.spring.cloud.service.implementation.identity.credential.provider.SpringTokenCredentialProvider.PASSWORDLESS_TOKEN_BEAN_NAME;
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jdbc/JdbcPropertiesBeanPostProcessor.java:27]: Shouldn't this be PASSWORDLESS_TOKEN_**CREDENTIAL**_BEAN_NAME?

**Review by saragluna — CHANGES_REQUESTED** [review]

**Comment by backwind1233** [comment]
Linke issue: https://github.com/Azure/azure-sdk-for-java/issues/30787

File: sdk/boms/spring-cloud-azure-dependencies/pom.xml [hunk 3]
```diff
@@ -55,6 +55,12 @@
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <!-- TODO (zhihaoguo): specify the version of azure-identity, we need to remove this before merging into main -->
```
Comment (saragluna) [inline sdk/boms/spring-cloud-azure-dependencies/pom.xml:58]: Pin the version of azure-identity to 1.6.0-beta.1 for we want to use the caching ability for MI.

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]

**Pull request merged**